### PR TITLE
Add Federation support

### PR DESCRIPTION
**Pull request by mattnorris**
Add Federation support via the `FEDERATION` environment variable. 

[Manual testing instructions](https://confluence-eng-gpk2.cisco.com/conf/display/WBXT/Federation)

### CODE REVIEW

File: packages/node_modules/@ciscospark/react-redux-spark/src/spark.js [hunk 1]
```diff
@@ -71,8 +71,13 @@ function defaultConfig(options = {}) {
     },
     credentials: {
       client_id: process.env.CISCOSPARK_CLIENT_ID,
+      federation: true,
```
Comment (adamweeks) [inline packages/node_modules/@ciscospark/react-redux-spark/src/spark.js]: Are we expected to have federation enabled for all users of the widgets?

File: packages/node_modules/@ciscospark/react-redux-spark/src/spark.js [hunk 2]
```diff
@@ -71,8 +71,13 @@ function defaultConfig(options = {}) {
...
       scope: 'spark:all spark:kms'
     },
+
+    // Federation EU cluster
+    orgId: process.env.EU_PRIMARY_ORG_ID,
```
Comment (adamweeks) [inline packages/node_modules/@ciscospark/react-redux-spark/src/spark.js]: I'm a bit confused by this env name. Are federation org ids all going to be "EU"?

File: package.json [hunk 3]
```diff
@@ -57,7 +57,9 @@
   "homepage": "https://github.com/webex/react-ciscospark#readme",
   "dependencies": {
     "@ciscospark/common": "1.49.3",
+    "@ciscospark/common-timers": "1.49.3",
```
Comment (adamweeks) [inline package.json]: Where are these used? Any idea why we need to specifically import them now?

File: package.json [hunk 4]
```diff
@@ -57,7 +57,9 @@
...
     "@ciscospark/helper-html": "1.49.3",
+    "@ciscospark/http-core": "1.49.3",
```
Comment (adamweeks) [inline package.json]: Could these work in devDependencies instead?

File: package.json [hunk 5]
```diff
@@ -45,10 +45,11 @@
...
   },
   "contributors": [
-    "Adam Weeks <adweeks@cisco.com>",
-    "Bernie Zang <nzang@cisco.com>",
+    "Arash Koushkebaghi <akoushke@cisco.com>",
     "Moriah Maney <momaney@cisco.com>",
-    "Arash Koushkebaghi <akoushke@cisco.com>"
+    "Matt Norris <matnorri@cisco.com> (http://mattnorris.me)",
+    "Adam Weeks <adweeks@cisco.com>",
+    "Bernie Zang <nzang@cisco.com>"
```
Comment (adamweeks) [inline package.json]: Is there a method to your sort order madness? 😆

Comment (mattnorris) [reply package.json]: Order of awesomeness 🎸

File: package.json (hunk 3 above)
Comment (mattnorris) [reply package.json]: There were several tests failing if I didn't include them. Specifically one of the files was the built dist of `spark-core`. It complained it couldn't find the module `common-timers` for example.

File: package.json (hunk 4 above)
Comment (mattnorris) [reply package.json]: I don't think so because tests were failing.

File: package.json (hunk 5 above)
Comment (adamweeks) [reply package.json]: Good call!

**Comment by adamweeks** [comment]
@mattnorris could you squash 0a0cff7 and ec326f0? That way we can have a cleaner changelog.

I'm assuming this isn't ready to go to master yet?

**Comment by mattnorris** [comment]
@adamweeks squashed. 

Is there a spot where widgets are hosted and point to Integration?

**Comment by adamweeks** [comment]
@mattnorris removing my approval since we discussed you were going to change this to be an env var controlled feature. Will also need a rebase on master due to jenkinsfile updates.

File: package.json [hunk 6]
```diff
@@ -45,36 +45,40 @@
...
-    "@ciscospark/internal-plugin-presence": "1.47.3",
-    "@ciscospark/internal-plugin-search": "1.47.3",
-    "@ciscospark/internal-plugin-team": "1.47.3",
-    "@ciscospark/internal-plugin-user": "1.47.3",
-    "@ciscospark/plugin-authorization": "1.47.3",
-    "@ciscospark/plugin-logger": "1.47.3",
-    "@ciscospark/plugin-people": "1.47.3",
-    "@ciscospark/plugin-phone": "1.47.3",
-    "@ciscospark/spark-core": "1.47.3",
+    "@ciscospark/common": "1.49.3",
```
Comment (mattnorris) [inline package.json]: @adamweeks should I bump this to the latest?

Comment (adamweeks) [reply package.json]: sure!

File: test/journeys/specs/smoke/multiple/index.js [hunk 7]
```diff
@@ -1,189 +0,0 @@
-import {assert} from 'chai';
```
Comment (adamweeks) [inline test/journeys/specs/smoke/multiple/index.js]: Why are we deleting this file?

Comment (mattnorris) [reply test/journeys/specs/smoke/multiple/index.js]: We are not... on purpose anyway.

**Pull request merged**